src/frontend/src/common/analytics/methods.ts

```diff
@@ -73,31 +84,64 @@ type EventValue = string | number | boolean | null;
  * expected in the Segment debugger before considering any work around
  * implementing or changing an event complete.
  *
- * All events include some generic, common info (CommonAnalyticsInfo). It
- * is possible to overwrite those with your `additionalEventData` if there
- * is a key collision. This is intentional -- some events could potentially
- * overrule the common info aspect -- but should be pretty unusual.
- *
- * Args:
- *   eventType (str, from EVENT_TYPES enum): canonical, unique name for event
- *   additionalEventData (obj): any additional data pertaining to event
- *     --- IMPORTANT USAGE NOTES ---
- *     The structure of this object will determine the (implicitly created)
- *     schema that Segment makes when it lands event in data warehouse. So:
- *     - All the keys in additionalEventData should be in **snake_case** to
- *     match expectation when working with analytics database.
- *     - additionalEventData should be a flat object of simple key-value pairs.
- *     You can send `undefined` for a value, but it will cause the key to be
- *     considered deleted for that event when it lands in Segment.
+ * When passing additionalEventData, it's best to directly construct it as an
+ * object in part of the function call to trigger "Excess Property Checking"
+ * and help ensure that the passed event data matches the event TS type being
+ * specified for the event. If instead you do it like this,
+ *     NOT GREAT  ---> analyticsTrackEvent<AnalyticsFooEvent>(
```

Comment:
can we avoid this with fancier typing?

Reply:
I do not believe so. I'm fairly confident that "Excess Property Checking" only happens in the context of an object literal. I spent a fair amount of time looking into the idea of an `Exact` type and as far as I could find, it does not exist. (There are possible workarounds, but when I tested them out I never was able to find a satisfactory approach.) You can see this GH issue on TS repo for what felt, to me at least, like the official "we're not going to do exact types": https://github.com/microsoft/TypeScript/issues/12936#issuecomment-284590083  -- it is a bit old from 2017, but I don't think the needle moved.

I spent way too much time poking at this, trying different approaches over a few days and searching the internet. This is the approach I was happiest with, although I'm still not in love with it.